symbolic/src/Data/Macaw/Symbolic.hs

```diff
@@ -182,33 +188,45 @@ import qualified Data.Macaw.Symbolic.MemTraceOps as MT
 -- The return value is a 'Maybe' so that architectures that do not yet support
 -- the translation can return 'Nothing', while still allowing fully generic client
 -- code to be written in terms of this class constraint.
-class ArchInfo arch where
-  archVals :: proxy arch -> Maybe (ArchVals arch)
+--
+-- The 'mem' type parameter indicates the memory model to be used for translation.
+-- The type alias 'ArchInfo' specializes this class to the llvm memory model
+-- using the 'LLVMMemory' type tag.
+class GenArchInfo mem arch where
+  genArchVals :: proxy mem -> proxy' arch -> Maybe (GenArchVals mem arch)
+
+type ArchInfo arch = GenArchInfo LLVMMemory arch
+
+archVals
+  :: ArchInfo arch
+  => proxy arch
+  -> Maybe (ArchVals arch)
+archVals p = genArchVals (Proxy @LLVMMemory) p
+
+data LLVMMemory
+
+class IsMemoryModel mem where
+  type MemModelType mem arch :: C.CrucibleType
+  type MemModelConstraint mem sym :: Constraint
```

Comment:
It looked like this might be the only use of the `constraints` package - can we just import `Constraint` from `GHC.Exts` instead?  That would also get rid of the import of Data.Constraints and the extra dependency in the cabal file.